### PR TITLE
Part 1: Average - string implementation

### PR DESCRIPTION
**Pull request by wildmaples**
# Changes
- Add the first bit of the Average test file that only introduces String
- Create and implement `StringConst` AST 
- Test on VMEmulator to ensure the VM code is correct

### CODE REVIEW

**Review by tomstuart — APPROVED** [review]
Yeah, this is excellent! A nice, clear, atomic change that gets us partway through implementing `Average`. Nice work!

I left a couple of nitpicks as usual, but nothing that should prevent you from merging this as-is if you want to.

File: lib/expression_parser.rb [hunk 1]
```diff
@@ -91,6 +91,18 @@ def write_vm_code(vm_writer, symbol_table)
   end
 end
 
+StringConst = Struct.new(:value) do
+  def write_vm_code(vm_writer, symbol_table)
+    vm_writer.write_push(:CONST, value.length)
+    vm_writer.write_call("String.new", 1)
+
+    value.bytes.each do |byte|
```
[the remark this answers is not in the document]
Comment (wildmaples) [reply lib/expression_parser.rb:99]: Thank you, this is informative. I remember learning this briefly when we worked on Mirth. I've also found [`String#codepoint`](https://ruby-doc.org/core-3.1.1/String.html#method-i-codepoints), which will return the same results.

[the remark this answers is not in the document]
Comment (tomstuart) [reply lib/expression_parser.rb]: Oh, I didn’t know about `#codepoints` and `#each_codepoint`! That’s even better.

[the remark this answers is not in the document]
Comment (tomstuart) [reply lib/expression_parser.rb]: For the record, we worked through https://github.com/tomstuart/utf-8-challenges and now the rabbit hole is complete 😂

**Pull request merged**